### PR TITLE
Add a .git-blame-ignore-revs

### PR DESCRIPTION
**Pull request by helen**
This adds a `.git-blame-ignore-revs` file to exclude certain commits from visual blame on GitHub.com, which can also be used on the command line via `git blame --ignore-revs-file .git-blame-ignore-revs`.

This is not a complete list of commits, though it does include all the commits I found for "pinking shears" with the exception of one that I noticed contained other changes. I would really love some help double checking these commits to make sure we're not including any that are anything besides whitespace/style fixes; it's better to be conservative to start. Would also like feedback / suggestions on the format of the file, particularly around the comments and how we label things to be able to maintain this file now and into the future.

Trac ticket: https://core.trac.wordpress.org/ticket/55422

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by costdev** [comment]
Some "Code is Poetry" commits (case insensitive search):
```bash
e4e1a54567dca12048e5344ab5b60a85e0d96743 # [4375]
e37aff4320b89bd94e21517f65d298f9f1fe6b37 # [45559]
```
and you already have this one under `# PHPCS`:
```bash
8f95800d52c1736d651ae6e259f90ad4a0db2c3f # [42343]
```

**Comment by helen** [comment]
@costdev I personally see those two commits as more than purely stylistic; I don't believe anything would have been affected by those specifically but sometimes subtle things like that (especially that final falsey check) can cause side effects and I would want to see those in the history. Definitely open to the idea that being more liberal about leaving commits out of our default code archaeology is a good idea, but I think I'd need a lot of convincing :)

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+48a3ec24c5ea7615431bde431098546674a0096c # [21070]
+2439e4d722f5093631c4bac9a45106ab33624aaa # [21381]
+56c4577feae624f7a3bf55d001295e0707c972b6 # [21486]
+680b671330327f2568549dec9460a2e942219d55 # [21492]
+dd39a3b3d2d1f792b6ed8e88b1fcc13877fbd8dd # [22491]
+6d8bce688f5215ad20133b2cefedc72ac5136be3 # [22634]
+05c0f14024a2bb8ef9bdfcb84589d04205f8ce24 # [23679]
+ece7a7714477142d4cdb3269d2f9ad3d5dab2e5f # [23780]
+687d1a2ce992a0e7e24675b506b85a57e42fe78e # [24303]
+45d2a20783460667f85d5fdced422efd54c33b72 # [24603]
```
Comment (peterwilsoncc) [inline .git-blame-ignore-revs:24]: Some formatting, mostly shears

File: .git-blame-ignore-revs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+c8889d984fd98381d6f0bd3228972e93348f1266 # [25085]
+21da24227f2c0087c95f44c60b1bca65cedf0611 # [25824]
```
Comment (peterwilsoncc) [inline .git-blame-ignore-revs]: Along with whitespace, also removes a `.` 

I hate me too sometimes...

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+48a504cc50a16f60a395ace8d133130e3d6962f5 # [30047]
+b539bd985bdf2dde162f46fcbf14e51e46ff8be7 # [30372]
+9a3942ffd82b28a77982e0a0d1c2bb0b8a70ce44 # [30996]
+469164785ff8defefbfcdf972c57d9004c273256 # [31077]
+c4e9c64233166957cd1cbcc54ff54b1a02edd3b5 # [31623]
+f4f1b4821342fd1d58708356b3ec39d6fefe31ac # [33411]
+0ec540b946eaebebc7e8ee39a8e4738914e7f2a3 # [33627]
+0c5bd752629b0960c6f43ca212a64724e4f40346 # [34534]
+991feb70438e981290696fa2b4fbe2ea54696a02 # [34774]
+6911ff11308089eace23719fc50160d403081a8e # [35627]
```
Comment (peterwilsoncc) [inline .git-blame-ignore-revs:46]: `nbsp` to actual space.

File: .git-blame-ignore-revs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+8df8cf2df14fe26174f97af5bb17d63f2e867231 # [42843]
+
+# PHPCS
```
Comment (peterwilsoncc) [inline .git-blame-ignore-revs]:
```suggestion
# Coding Standards
```

A nit. I think it fine to group PHP, JS and CSS coding standards fixes in the one spot.

**Review by peterwilsoncc — APPROVED** [review]
Looks good to me.

Commits reviewed with whitespace suppressed, I've noted pinking shears commits with more than white space changes. 

The final commit I've not reviewed in full as it was a `phpcbf` run and Stephen and I were staring checked Gary's work at the time.

**Comment by peterwilsoncc** [comment]
> I personally see those two commits as more than purely stylistic; I don't believe anything would have been affected by those specifically but sometimes subtle things like that (especially that final falsey check)

Yeah, e37aff4320b89bd94e21517f65d298f9f1fe6b37 broke something and was caught in the following commit. My code was not poetry.

File: .git-blame-ignore-revs (hunk 1 above)
Comment (helen) [reply .git-blame-ignore-revs:24]: This formatting seems okay to me

File: .git-blame-ignore-revs (hunk 2 above)
Comment (helen) [reply .git-blame-ignore-revs]: Sigh. Should probably exclude this commit then. What do you think of leaving it in here so if in the future people auditing this for some ghastly reason aren't left wondering if this was missed originally?

```suggestion
# 21da24227f2c0087c95f44c60b1bca65cedf0611 # [25824] includes a punctuation change
```

File: .git-blame-ignore-revs (hunk 3 above)
Comment (helen) [reply .git-blame-ignore-revs:46]: That seems fine to me

File: .git-blame-ignore-revs (hunk 4 above)
Comment (helen) [reply .git-blame-ignore-revs]: Sure, maybe if/as it grows there can be subheadings or split back out.

**Comment by helen** [comment]
> > I personally see those two commits as more than purely stylistic; I don't believe anything would have been affected by those specifically but sometimes subtle things like that (especially that final falsey check)
> 
> Yeah, [e37aff4](https://github.com/WordPress/wordpress-develop/commit/e37aff4320b89bd94e21517f65d298f9f1fe6b37) broke something and was caught in the following commit. My code was not poetry.

@peterwilsoncc I feel both disappointed in myself that I didn't actually check on [follow up commits](https://github.com/WordPress/wordpress-develop/commit/6ee8a99b6a8712d6f025003a084e496626673f22) and weirdly gratified that my worst instincts remain correct 😆

File: .git-blame-ignore-revs (hunk 2 above)
Comment (peterwilsoncc) [reply .git-blame-ignore-revs]: > What do you think of leaving it in here so if in the future people auditing this for some ghastly reason aren't left wondering if this was missed originally?

Works for me, I've committed the suggestion.

**Comment by helen** [comment]
Committed in 921d2805a0ebaf4db9f75ea9f5af46e48d6f68ac

**Pull request closed without merging**